### PR TITLE
KOOK-324 Retain text formatting for highlights/annotation across elem…

### PR DESCRIPTION
**Pull request by subzthanabalan**
…ents

https://safari.atlassian.net/browse/KOOK-324

Saving the text selection html as a fragment property and passing that back to the Android app.

### CODE REVIEW

File: src/ui/textselector.js [hunk 1]
```diff
@@ -70,6 +70,12 @@ TextSelector.prototype.captureDocumentSelection = function () {
             browserRange = new xpathRange.Range.BrowserRange(r),
             normedRange = browserRange.normalize().limit(this.element);
 
+        // Grab the html from text selection, and add it as a fragment property
+        var documentFragment = r.cloneContents();
+        var elementContainer = document.createElement("div");
+        elementContainer.appendChild(documentFragment);
+        normedRange.fragment = elementContainer.innerHTML;
```
Comment (moagrius) [inline src/ui/textselector.js]: oh yeah ha this a very common trick we have to use in js, probably should have mentioned it - did this cause you any trouble or did you figure it out right away?  or did you already know it?

File: src/ui/main.js [hunk 2]
```diff
@@ -33,20 +33,25 @@ exports.main = function(options) {
...
      var selector = '.' + className;
      for (var i = 0, len = ranges.length; i < len; i++) {
          var range = ranges[i];
          text.push(range.text().trim());
+         fragment.push(range.fragment);
          serializedRanges.push(range.serialize(options.element, selector));
      }
 
      text = text.join(' / ');
+     fragment = fragment.join(' ');
```
Comment (moagrius) [inline src/ui/main.js]: presumably this is for the concept of "multiple ranges" that don't exist now, so this is just to be future-proof?

**Comment by moagrius** [comment]
looks good +1

File: src/ui/textselector.js (hunk 1 above)
Comment (subzthanabalan) [inline src/ui/textselector.js]: Took me a little bit to figure out, super handy!

File: src/ui/main.js (hunk 2 above)
Comment (subzthanabalan) [inline src/ui/main.js]: Yeah

**Pull request merged**